### PR TITLE
Fix header swapping and forwarding back

### PR DESCRIPTION
**Pull request by ebiiii**

### CODE REVIEW

File: app/gateway/proxy.py [hunk 1]
```diff
@@ -80,35 +81,39 @@ def pass_through(path):
     # TODO: This switch should be resource dependent.
     if path.startswith('storage'):
         url = app.config['RENKU_ENDPOINT'] + "/api/" + path
+        auth_headers = headers
     else:
         url = app.config['GITLAB_URL'] + "/api/" + path
```
Comment (3C111) [inline app/gateway/proxy.py]: Why is this part moved in the 'else' clause? Is there no need for authentication on the storage part?

**Comment by ebiiii** [comment]
I close this one and start a new branch on it to implement the generic configuration for endpoints.
